### ISSUE TITLE
run selection/line in python terminal not working

### ISSUE THREAD

**Issue by MarkoNovi** [microsoft/vscode#184681]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.79.0
- OS Version: Windows 10

Steps to Reproduce:

1. select couple of lines in python file
2. press SHIFT + ENTER, or right click selection -> Run Python -> run selection/line in python terminal

this started happening after last update.

**Comment by pkoprov** [comment]
I have a similar issue:
when I press SHIFT + ENTER, or right click selection -> Run Python -> run selection/line in the python terminal it opens 2 terminals named Python, where the first one opens the python terminal (PID 29056) but the second  (PID 434444) does not. After that, it tries to execute the highlighted code in the second terminal which obviously is yielding an error.
If I kill the second Python terminal  (PID 434444) and try to run a line of Python again, it opens a new terminal with a new PID and no Python console and behaves exactly like the second one.
If I kill the first terminal and then run the line it does not create another console, but still not running the code.
Finally, if I type `python` in the second console it opens the python terminal, and if I start running line-by-line execution it works.

**Comment by illmattic-1** [comment]
I have the exact same issue.

Thank you for providing a work around

**Comment by luis0283** [comment]
> I have a similar issue: when I press SHIFT + ENTER, or right click selection -> Run Python -> run selection/line in the python terminal it opens 2 terminals named Python, where the first one opens the python terminal (PID 29056) but the second (PID 434444) does not. After that, it tries to execute the highlighted code in the second terminal which obviously is yielding an error. If I kill the second Python terminal (PID 434444) and try to run a line of Python again, it opens a new terminal with a new PID and no Python console and behaves exactly like the second one. If I kill the first terminal and then run the line it does not create another console, but still not running the code. Finally, if I type `python` in the second console it opens the python terminal, and if I start running line-by-line execution it works.

The same thing happens to me since yesterday, thank you very much for the solution!!!, and I hope they fix it later.

**Comment by Beenjamming** [comment]
Exact same issue.  Workaround also works but didn't have the issue till yesterday.

**Comment by VincentGurgul** [comment]
Where did this bug suddenly come from? I don't remember installing any updates of VS Code. Does VS Code update automatically in the background? If so, why can't one go back to an older version, like with the extensions, if there is a serious bug like this that isn't being addressed? Makes me want to switch back to vim. 😡

**Comment by Beenjamming** [comment]
> 

Vincent,

It is still open.  I closed my issue as it was a duplicate and linked to this one.  You can roll back VS code: https://code.visualstudio.com/updates/v1_78  

I installed the April 2023 release and the issue is not present.  You will also need to roll back the python extension.  Hope this helps!

**Comment by VincentGurgul** [comment]
Version 1.78.2 had the same bug for me (Mac silicon / arm64). Had to switch back to 1.77.3, now it is working again. Thanks a lot for the link.

**Comment by Beenjamming** [comment]
Glad 1.77.3 worked!  Also wanted to mention that you may need to change your default update settings under your profile settings to prevent auto updates.

**Comment by VincentGurgul** [comment]
Good point, I just reopened the program and it had automaically reinstalled 1.79.0 ... buuuut: the issue wasn't there 😮 So I checked and I still had the Python extension v2023.8.0 running. Apparently the bug is caused by v2023.10.0!

**Comment by AyoubAarab** [comment]
> Good point, I just reopened the program and it had automaically reinstalled 1.79.0 ... buuuut: the issue wasn't there 😮 So I checked and I still had the Python extension v2023.8.0 running. Apparently the bug is caused by v2023.10.0!
> 
> So downgrading the Python extension is the solution to the problem.

I have tried this myself and can confirm the issue is caused by the v2023.10.0 of the Python VSCode extension. Downgrading to v2023.8.0 solves the issue.

**Comment by Peter-Hegarty-Data** [comment]
The issue resolved for me by just rolling back the Python extension to v2023.8.0.
